lib/flowViewport/selection/fvpPathMapperRegistry.cpp

```diff
@@ -101,28 +112,80 @@ PathMapperConstPtr PathMapperRegistry::GetMapper(const Ufe::Path& path) const
     }
     // We reached the end of the parent path without returning true, therefore
     // there are no ancestors.
-    return fallbackMapper;
+    return nullptr;
 }
 
-Fvp::PrimSelections ufePathToPrimSelections(const Ufe::Path& appPath)
+PathMapperConstPtr PathMapperRegistry::GetMapper(const Ufe::Path& path) const
 {
-    PXR_NAMESPACE_USING_DIRECTIVE
+    if (path.empty()) {
+        return nullptr;
+    }
 
-    Fvp::PrimSelections primSelections;
+    auto mapper = _GetMapper(path);
+    return mapper ? mapper : fallbackMapper;
+}
 
-    auto mapper = Fvp::PathMapperRegistry::Instance().GetMapper(appPath);
+PrimSelections PathMapperRegistry::UfePathToPrimSelections(
+    const Ufe::Path& appPath
+)
+{
+    PrimSelections primSelections;
+
+    auto mapper = GetMapper(appPath);
         
     if (!mapper) {
         TF_WARN("No registered mapping for path %s, no prim path returned.", Ufe::PathString::string(appPath).c_str());
     }
     else {
         primSelections = mapper->UfePathToPrimSelections(appPath);
         if (primSelections.empty()) {
-            TF_WARN("Mapping for path %s returned no prim path.", Ufe::PathString::string(appPath).c_str());
+
+            mapper = GetFallbackMapper();
+        
+            if (!mapper) {
+                TF_WARN("No registered fallback mapping, no prim path returned.");
+            }
+            else {
+                primSelections = mapper->UfePathToPrimSelections(appPath);
+                if (primSelections.empty()) {
+                    TF_WARN("Mapping for path %s returned no prim path.", Ufe::PathString::string(appPath).c_str());
+                }
+            }
         }
     }
 
     return primSelections;
 }
 
+bool PathMapperRegistry::HasMapper(const Ufe::Path& path) const
+{
+    return _GetMapper(path) != nullptr;
```

Comment:
I would add a :
 if (path.empty()) {
        return nullptr;
    }
    
    As if path is empty it will crash in _GetMapper with TF_AXIOM.